aten/src/ATen/native/mps/operations/View.mm

```diff
@@ -417,23 +417,49 @@
   return outputTensor;
 }
 
-MPSGraphTensorData* getMPSGraphTensorDataForView(const Tensor& src, MPSShape *mpsShape, const MPSDataType mpsDataType) {
-  IntArrayRef src_base_shape = get_buffer_shape(src.storage().data());
-  std::vector<int64_t> src_view_shape;
+static
+std::vector<int64_t> getViewShape(const Tensor& src, MPSShape *mpsShape) {
   bool hasMPSShape = (mpsShape != nil);
-  int src_ndim_base = src_base_shape.size();
-  int src_ndim_view = 0;
+  std::vector<int64_t> src_view_shape;
   if (hasMPSShape) {
-    src_ndim_view = [mpsShape count];
-    src_view_shape.reserve(src_ndim_view);
+    int src_ndim_view = [mpsShape count];
+    src_view_shape.resize(src_ndim_view);
     for (const auto i : c10::irange(src_ndim_view)) {
       src_view_shape[i] = [mpsShape[i] intValue];
     }
   } else {
-    src_ndim_view = src.dim();
     src_view_shape = src.sizes().vec();
   }
 
+  return src_view_shape;
+}
+
+bool canSliceViewTensor(const Tensor& src, MPSShape *mpsShape) {
```

Comment:
I wonder if this code wasn't used before to check the Slice?

Reply:
Is there a core function which can detect Slice pattern?

Reply:
hmm, I'll take a look to see if what we have in View.mm matches with this code. In canSliceViewTensor I wanted to add all the cases that cannot be solved through the arrayView API